e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -65,5 +69,112 @@ internal static class TestHelper {
+            await _registryManager.UpdateTwinAsync(twin.DeviceId, patch, twin.ETag);
+        }
+
+        /// <summary>
+        /// transfer the content of published_nodes.json file into the OPC Publisher edge module
+        /// </summary>       
+        /// <param name="sourceFilePath">Source file path</param>
+        /// <param name="destinationFilePath">Destination file path</param
+        public static void LoadPublishedNodesFile(string sourceFilePath, string destinationFilePath) {
+            var username = Environment.GetEnvironmentVariable("PCS_SIMULATION_USER");
+            var password = Environment.GetEnvironmentVariable("PCS_SIMULATION_PASSWORD");
+            var host = Environment.GetEnvironmentVariable("IOT_EDGE_DEVICE_DNS_NAME");
+           
+            Assert.True(!string.IsNullOrWhiteSpace(username), "username string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(password), "password string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(host), "host string is null");
+
+            using (ScpClient client = new ScpClient(host, username, password)) {
+                client.Connect();
+
+                if (string.IsNullOrEmpty(sourceFilePath)) {
+                    DeletePublishedNodesFile(destinationFilePath);
+                }
+                using (Stream localFile = File.OpenRead(sourceFilePath)) {
+                    client.Upload(localFile, destinationFilePath);
+                }
+                client.Disconnect();
+            }
+        }
+
+        /// <summary>
+        /// Delete published_nodes.json file into the OPC Publisher edge module
+        /// </summary>       
+        /// <param name="sourceFilePath">Source file path</param>
+        /// <param name="destinationFilePath">Destination file path</param
+        public static bool DeletePublishedNodesFile(string destinationFilePath) {
+            var username = Environment.GetEnvironmentVariable("PCS_SIMULATION_USER");
+            var password = Environment.GetEnvironmentVariable("PCS_SIMULATION_PASSWORD");
+            var host = Environment.GetEnvironmentVariable("IOT_EDGE_DEVICE_DNS_NAME");
+            
+            Assert.True(!string.IsNullOrWhiteSpace(username), "username string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(password), "password string is null");
+            Assert.True(!string.IsNullOrWhiteSpace(host), "host string is null");
+
+            var isSuccessful = false;
+            using (SshClient client = new SshClient(host, username, password)) {
+                client.Connect();
+                var terminal = client.RunCommand("rm " + destinationFilePath);
+                if (string.IsNullOrEmpty(terminal.Error)) {
+                    isSuccessful = true;
+                }
+                client.Disconnect();
+            }
+            return isSuccessful;
+        }
+
+        private static RegistryManager _registryManager;
```

Comment:
>private static RegistryManager _registryManager; [](start = 8, length = 48)

Instead of using private static member, please create a method "CreateIoTHubContext" that return a context object that encapsulate the access to registry manager, this context object is than a mandatory parameter for "SwitchToStandaloneMode", "SwitchToOrchestratedMode", "UpdateTagAsync"

Reply:
done

Reply:
Karapet in this [PR](https://github.com/Azure/Industrial-IoT/pull/911/files) has added a RegistryHelper. I can use that.